src/new_hope/week1.clj

```diff
@@ -48,5 +48,116 @@
   ;; More elaborate benchmark taking 10 seconds
   (let [coll (vec (range 10000))]
     (crit/quick-bench (second (reverse coll)))
-    (crit/quick-bench (last (butlast coll))))
-  )
+    (crit/quick-bench (last (butlast coll)))))
+
+
+;; http://www.4clojure.com/problem/24
+(def sum-it-all-up
+  (fn [coll]
+    (apply + coll)))
+
+(facts "about sum-it-all-up"
+       (sum-it-all-up [1 2 3]) => 6
+       (sum-it-all-up (list 0 -2 5 5)) => 8
+       (sum-it-all-up #{4 2 1}) => 7
+       (sum-it-all-up '(0 0 -1)) => -1
+       (sum-it-all-up '(1 10 3)) => 14)
+
+;;http://www.4clojure.com/problem/25
+(def filter-odd
+  (fn [coll]
+    (filter odd? coll)))
+
+(facts "about filter-odd"
+       (filter-odd #{1 2 3 4 5}) => '(1 3 5)
+       (filter-odd [4 2 1 6]) => '(1)
+       (filter-odd [2 2 4 6]) => '()
+       (filter-odd [1 1 1 3]) => '(1 1 1 3))
+
+;;http://www.4clojure.com/problem/27
+(def palindrome?
+  (fn [coll]
+    (= (apply str (reverse coll)) (apply str coll))))
+
+(facts "about palindrome-detect"
+       (palindrome? '(1 2 3 4 5)) => false
+       (palindrome? "racecar") => true
+       (palindrome? [:foo :bar :foo]) => true
+       (palindrome? '(1 1 3 3 1 1)) => true
+       (palindrome? '(:a :b :c)) => false)
+
+;;http://www.4clojure.com/problem/32
+(def duplicate-a-seq
+  (fn [coll] (sort (apply conj coll coll))))
+
+(facts "about duplicate-a-seq"
+       (duplicate-a-seq [1 2 3]) => '(1 1 2 2 3 3)
+       (duplicate-a-seq [:a :a :b :b]) => '(:a :a :a :a :b :b :b :b)
+       (duplicate-a-seq [[1 2] [3 4]]) => '([1 2] [1 2] [3 4] [3 4])
+       (duplicate-a-seq [[1 2] [3 4]]) => '([1 2] [1 2] [3 4] [3 4]))
+
+;; Day 5
+;;http://www.4clojure.com/problem/30
+(def compress-seq
+  (fn [coll]
+    (loop [in coll
+           out []]
+      (if (not (seq in))
```

Comment:
it's better to use `(if (seq s) ...)` instead of `(if (not (empty? s)) ...)`
but with `(if (empty? s) ...)` certainly looks better than `(if (not (seq s)) ...)`
`(empty? s)` is defined exactly as `(not (seq s))` btw